lib/util/ride_event.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../rides/models/ride.dart';
+import 'model.dart';
+import 'supabase.dart';
+
+class RideEvent extends Model {
+  RideEventCategory category;
+  bool read;
+  final int rideId;
+  final Ride? ride;
+
+  RideEvent({
+    super.id,
+    super.createdAt,
+    required this.category,
+    this.read = false,
+    required this.rideId,
+    this.ride,
+  });
+
+  factory RideEvent.fromJson(Map<String, dynamic> json) {
+    return RideEvent(
+      id: json['id'],
+      createdAt: DateTime.parse(json['created_at']),
+      rideId: json['ride_id'],
+      ride: json.containsKey('ride') ? Ride.fromJson(json['ride']) : null,
+      category: RideEventCategory.values.elementAt(json['category'] as int),
+      read: json['read'],
+    );
+  }
+
+  static List<RideEvent> fromJsonList(List<Map<String, dynamic>> jsonList) {
+    return jsonList.map((Map<String, dynamic> json) => RideEvent.fromJson(json)).toList();
+  }
+
+  @override
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'ride_id': rideId,
+      'category': category.index,
+      'read': read,
+    };
+  }
+
+  Future<void> markAsRead() async {
+    read = true;
+    //custom rpc call to mark rideEvent as read, so the user does not need the write permission on the ride_events table
+    await SupabaseManager.supabaseClient.rpc('mark_ride_event_as_read', params: <String, dynamic>{'ride_event_id': id});
+  }
+
+  bool isForCurrentUser() {
+    final int profileId = SupabaseManager.getCurrentProfile()!.id!;
+    if (category == RideEventCategory.approved ||
+        category == RideEventCategory.cancelledByDriver ||
+        category == RideEventCategory.rejected) {
+      return ride!.rider!.id == profileId;
```

Comment:
riderId and driverId below would be simpler

Reply:
changed it to rider/driver.isCurrentUser